src/Symfony/Component/Serializer/Mapping/ClassDiscriminatorMapping.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Mapping;
+
+/**
+ * @author Samuel Roze <samuel.roze@gmail.com>
+ */
+final class ClassDiscriminatorMapping
+{
+    private $typeProperty;
+    private $typesMapping;
+
+    public function __construct(string $typeProperty, array $typesMapping = array())
+    {
+        $this->typeProperty = $typeProperty;
+        $this->typesMapping = $typesMapping;
+    }
+
+    public function getTypeProperty(): string
+    {
+        return $this->typeProperty;
+    }
+
+    public function addTypeMapping(string $type, string $typeClass)
```

Comment:
Is it possible to remove this method? In the PropertyInfo component we used immutable objects.